### PR TITLE
Features/openapi v3

### PR DESCRIPTION
**Pull request by WiredSharp**
[WIP] ApiModel

### CODE REVIEW

File: Clients/src/Swag4Net.Core/ApiModel.fs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      License:License option }
+  and Contact = 
+    | Email of string
+  and License = 
+    { Name:string; Url:string }
+  and Schema =
+      {
+        Type: DataType
+        Title: string option
+        AllOf: Schema option
```
Comment (rflechner) [inline Clients/src/Swag4Net.Core/ApiModel.fs:31]: AllOf is a list

File: Clients/src/Swag4Net.Core/ApiModel.fs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        ExclusiveMinimum: int option
+        MaxLength: int option
+        MinLength: int option
+        Pattern: RegularExpression option
+        MaxItems: int option
+        MinItems: int option
+        UniqueItems: bool option
+        MaxProperties: int option
+        MinProperties: int option
+        Properties: Map<string, Schema> option
```
Comment (rflechner) [inline Clients/src/Swag4Net.Core/ApiModel.fs:49]: Please use `PropertyType` 
https://github.com/rflechner/Swag4Net-1/blob/master/Clients/src/ClientsForSwagger.Core/Models.fs#L32
In domain, property type are resolved and discriminated

File: Clients/src/Swag4Net.Core/ApiModel.fs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        AdditionalProperties: AdditionalProperties option
+        Required: bool option
+        Nullable: bool option
+        Enum: Any list option
+        Format: DataTypeFormat option
```
Comment (rflechner) [inline Clients/src/Swag4Net.Core/ApiModel.fs:54]: DataTypeFormat should be a discriminated union

File: Clients/src/Swag4Net.Core/v3/SpecificationDocument.fs [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
 module SpecificationDocument =
 
   type TypeName = string
+  type Anchor = Anchor of string
   type HttpStatusCode = string
-  type TypeOrReference<'a> =
-     | T of 'a
-     | R of Reference
-  and Reference = string
+  type TypeOrReference<'TTarget> =
```
Comment (rflechner) [inline Clients/src/Swag4Net.Core/v3/SpecificationDocument.fs]:
```
type InlinedOrReferenced<'a> =
     | Inlined of 'a
     | Referenced of Reference
```
'Inline or referenced' are the terms used by OpenApi

File: Clients/src/Swag4Net.Core/ApiModel.fs (hunk 1 above)
Comment (WiredSharp) [reply Clients/src/Swag4Net.Core/ApiModel.fs:31]: En effet, comme les autres xxOf d'ailleurs

File: Clients/src/Swag4Net.Core/ApiModel.fs (hunk 3 above)
Comment (WiredSharp) [reply Clients/src/Swag4Net.Core/ApiModel.fs:54]: it is too restrictive to use a discriminated union here as format list might be extended

File: Clients/src/Swag4Net.Core/ApiModel.fs (hunk 2 above)
Comment (WiredSharp) [reply Clients/src/Swag4Net.Core/ApiModel.fs:49]: don't really understand what you mean, property are just tagged association between two known types

**Comment by WiredSharp** [comment]
i merged modifications from [openv3_parsing](https://github.com/rflechner/Swag4Net-1/feature/openv3_parsing)

**Pull request closed without merging**